portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -66,19 +73,44 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     /** Servlet that adds the comment data to the datastore */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Get the input from the form.
-        String commentText = request.getParameter("comment");
-        long timestamp = System.currentTimeMillis();
-
-        Entity commentEntity = new Entity("commentEntity");
-        commentEntity.setProperty("comment", commentText);
-        commentEntity.setProperty("timestamp", timestamp);
-        
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(commentEntity);
 
-        // Respond with the result.
-        response.sendRedirect("/#comments");
+        UserService userService = UserServiceFactory.getUserService();
+        if (userService.isUserLoggedIn()) {
+            // Get the input from the form.
+            String commentText = request.getParameter("comment");
+            String nickname = request.getParameter("nickname");
+
+            long timestamp = System.currentTimeMillis();
+
+            Entity commentEntity = new Entity("commentEntity");
+            commentEntity.setProperty("comment", commentText);
+            commentEntity.setProperty("timestamp", timestamp);
+            commentEntity.setProperty("nickname", nickname);
+
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            datastore.put(commentEntity);
+
+
+            response.sendRedirect("/#comments");
+
+        }
+        else{
+            response.sendRedirect("/#comments");
+        }
+
+    }
+
+
+    @Override
+    public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        if (userService.isUserLoggedIn()) {
```

Silent failure is not ideal.  It would be better to return a 403 error and catch that on the UI side of things.